src/app/api/newLoan/route.ts

```diff
@@ -0,0 +1,185 @@
+import { NextResponse, NextRequest } from 'next/server';
+import LoanModel from '@/models/loan-model';
+
+interface LoanRequest {
+    userAddress: string;
+    loanAmount: string;
+    loanToken: string;
+    collateralAmount: string;
+    collateralToken: string;
+    loanPeriod: string;
+    loanRequestPeriod: string;
+    healthFactor: string;
+    platformFee: string;
+    interestRate: string;
+    initialThreshold: string;
+    liquidationThreshold: string;
+    nftManager: string;
+    nftVersion: string;
+}
+
+export async function POST(request: NextRequest) {
+    try {
+        const {
+            userAddress,
+            loanAmount,
+            loanToken,
+            collateralAmount,
+            collateralToken,
+            loanPeriod,
+            loanRequestPeriod,
+            healthFactor,
+            platformFee,
+            interestRate,
+            initialThreshold,
+            liquidationThreshold,
+            nftManager,
+            nftVersion,
+        }: LoanRequest = await request.json();
+
+        if (!userAddress) {
+            return NextResponse.json({
+                status: 400,
+                message: 'ERROR: Wallet Address is required.',
+            });
+        }
+
+        if (!loanAmount) {
+            return NextResponse.json({
+                status: 400,
+                message: 'ERROR: Loan Amount is required.',
+            });
+        }
+
+        if (!loanToken) {
+            return NextResponse.json({
+                status: 400,
+                message: 'ERROR: Loan Token is required.',
+            });
+        }
+
+        if (!collateralAmount) {
+            return NextResponse.json({
+                status: 400,
+                message: 'ERROR: Collateral Amount is required.',
+            });
+        }
+
```

Comment:
This is crazy, We need to use a parameter validation library 

https://dev.to/meddlesome/nextjs-apis-validator-with-middleware-3njl